google/cloud/spanner_dbapi/cursor.py

```diff
@@ -4,246 +4,103 @@
-
-        if parts.get("homogenous"):
-            # The common case of multiple values being passed in
-            # non-complex pyformat args and need to be uploaded in one RPC.
-            return self._connection.in_transaction(
-                self.__do_execute_insert_homogenous, parts
-            )
-        else:
-            # All the other cases that are esoteric and need
-            #   transaction.execute_sql
-            sql_params_list = parts.get("sql_params_list")
-            return self._connection.in_transaction(
-                self.__do_execute_insert_heterogenous, sql_params_list
-            )
-
-    def __do_execute_insert_heterogenous(self, transaction, sql_params_list):
-        for sql, params in sql_params_list:
-            sql, params = sql_pyformat_args_to_spanner(sql, params)
-            param_types = get_param_types(params)
-            res = transaction.execute_sql(
-                sql, params=params, param_types=param_types
-            )
-            # TODO: File a bug with Cloud Spanner and the Python client maintainers
-            # about a lost commit when res isn't read from.
-            _ = list(res)
-
-    def __do_execute_insert_homogenous(self, transaction, parts):
-        # Perform an insert in one shot.
-        table = parts.get("table")
-        columns = parts.get("columns")
-        values = parts.get("values")
-        return transaction.insert(table, columns, values)
-
-    def __handle_DQL(self, sql, params):
-        with self._connection.read_snapshot() as snapshot:
-            # Reference
-            #  https://googleapis.dev/python/spanner/latest/session-api.html#google.cloud.spanner_v1.session.Session.execute_sql
-            sql, params = sql_pyformat_args_to_spanner(sql, params)
-            res = snapshot.execute_sql(
-                sql, params=params, param_types=get_param_types(params)
-            )
-            if type(res) == int:
-                self._row_count = res
-                self._itr = None
-            else:
-                # Immediately using:
-                #   iter(response)
-                # here, because this Spanner API doesn't provide
-                # easy mechanisms to detect when only a single item
-                # is returned or many, yet mixing results that
-                # are for .fetchone() with those that would result in
-                # many items returns a RuntimeError if .fetchone() is
-                # invoked and vice versa.
-                self._res = res
-                # Read the first element so that StreamedResult can
-                # return the metadata after a DQL statement. See issue #155.
-                self._itr = PeekIterator(self._res)
-                # Unfortunately, Spanner doesn't seem to send back
-                # information about the number of rows available.
-                self._row_count = _UNSET_COUNT
+    @property
+    def connection(self):
+        return self._connection
```

Comment:
We could just write `self.connection = connection` in init. For now there is not need in property definition